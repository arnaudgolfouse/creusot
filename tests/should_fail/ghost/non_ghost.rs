extern crate creusot_contracts;
use creusot_contracts::*;

#[check(terminates)]
fn terminating() {}

pub fn calls_terminating() {
    ghost! {
        terminating();
    };
}
