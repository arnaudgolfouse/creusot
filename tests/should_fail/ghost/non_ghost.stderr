error: called non-ghost function `terminating` in ghost context
 --> non_ghost.rs:9:9
  |
9 |         terminating();
  |         ^^^^^^^^^^^

error: aborting due to 1 previous error

